src/Model.php

```diff
@@ -609,37 +625,50 @@ public function isDirty(string $field): bool
      *
      * @return Field[]
      */
-    public function getFields($filter = null): array
+    public function getFields($filters = null): array
     {
-        if ($filter === null) {
+        if ($filters === null) {
             return $this->fields;
-        } elseif (is_string($filter)) {
-            $filter = [$filter];
+        } elseif (!is_array($filters)) {
+            $filters = [$filters];
         }
 
-        return array_filter($this->fields, function (Field $field, $name) use ($filter) {
-            // do not return fields outside of "only_fields" scope
-            if ($this->only_fields && !in_array($name, $this->only_fields, true)) {
-                return false;
-            }
-            foreach ($filter as $f) {
-                if (
-                    ($f === 'system' && $field->system)
-                    || ($f === 'not system' && !$field->system)
-                    || ($f === 'editable' && $field->isEditable())
-                    || ($f === 'visible' && $field->isVisible())
-                ) {
+        return array_filter($this->fields, function (Field $field, $name) use ($filters) {
+            foreach ($filters as $filter) {
+                if ($this->fieldMatchesFilter($field, $filter)) {
                     return true;
```

Comment:
this imply OR - do we want that and support array at all? I think that is illogical, we should probably drop array support, wdyt?

Reply:
This is how it was, no changes. I suggest to keep it like it is.

Reply:
I highly prefer to address the topic completely when opened.

Reply:
I see no problem with results being OR

Reply:
but current/expected use case?